src/main/java/de/caritas/cob/agencyservice/api/admin/service/agency/DemographicsConverter.java

```diff
@@ -0,0 +1,76 @@
+package de.caritas.cob.agencyservice.api.admin.service.agency;
+
+import static org.springframework.util.StringUtils.isEmpty;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Lists;
+import de.caritas.cob.agencyservice.api.model.AgencyAdminFullResponseDTO;
+import de.caritas.cob.agencyservice.api.model.DemographicsDTO;
+import de.caritas.cob.agencyservice.api.repository.agency.Agency;
+import de.caritas.cob.agencyservice.api.repository.agency.Agency.AgencyBuilder;
+import de.caritas.cob.agencyservice.api.repository.agency.Gender;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+/**
+ * Builder to build an {@link AgencyAdminFullResponseDTO()} from an {@link Agency} instance.
+ */
+@RequiredArgsConstructor
+@Component
+public class DemographicsConverter {
+
+  public DemographicsDTO convertToDTO(Agency agency) {
+    return new DemographicsDTO().ageTo(nullSafeToInteger(agency.getAgeTo()))
+        .ageFrom(nullSafeToInteger(agency.getAgeFrom())).genders(getGenderNames(agency));
+  }
+
+  public void convertToEntity(DemographicsDTO demographicsDTO, AgencyBuilder agencyBuilder) {
+    convertGender(demographicsDTO, agencyBuilder);
+    agencyBuilder.ageTo(toShort(demographicsDTO.getAgeTo()));
+    agencyBuilder.ageFrom(toShort(demographicsDTO.getAgeFrom()));
+  }
+
+  private Short toShort(Integer age) {
```

Comment:
if you like "nullSafeToInteger" then we should go with "nullSafeToShort"

Reply:
done